trussworks/react-uswds
# fix: disable postinstall script prior to publishing

# Note!!:
The alternative to this is to roll back husky to v4: https://github.com/trussworks/react-uswds/pull/926. We should merge one and close the other + delete the branch.

Due to Husky v5 currently causing more headaches than it may be worth, I recommend rolling back to v4 (PR linked above)

# Summary

Husky 5's implementation is a bit different and required adding a `postinstall` script to package.json. This script should only be ran while developing, not for users of the packaged artifact.

Solution is to disable post install scripts before publishing. This is what `pinst` was added for in https://github.com/trussworks/react-uswds/pull/867, however I missed adding the disable. (See [pinst usage ](https://www.npmjs.com/package/pinst#usage) for more info)

closes: https://github.com/trussworks/react-uswds/issues/923

## How To Test

1. Checkout this branch:
```
git fetch
git checkout bl-fix-husky-packaged-post-install-923
```
2. Perform a clean install to verify dev still works (I deleted `node_modules` and ran `yarn` to verify, plus the ability to commit this and see the git hooks still working with husky were good tests)

3. Locally running `yarn prepublishOnly` should run the existing test and linting scripts as well as disabling `postInstall`:
![image](https://user-images.githubusercontent.com/15805554/108557090-051fe700-72c6-11eb-914c-b10dd438dbdb.png)
![image](https://user-images.githubusercontent.com/15805554/108557130-1668f380-72c6-11eb-8478-b5b74a1a5aeb.png)

5. locally running `yarn postpublish` should result in postInstall being enabled once again:
![image](https://user-images.githubusercontent.com/15805554/108557236-47492880-72c6-11eb-9205-2c5dc3251549.png)
![image](https://user-images.githubusercontent.com/15805554/108557264-55974480-72c6-11eb-8521-00ea2f353c77.png)

6. Finally, testing an install of this package after release of this code change will be the true test.
